### PR TITLE
making STFT a simple wrapper for torch.stft + some var name changes

### PR DESCRIPTION
**Pull request by keunwoochoi**
Seems like there's no reason for non-zero constant padding, which means we could already make our STFT a  wrapper for `torch.stft`. 
I also removed the old default values (e.g.. `fft_len=2048`), I don't think we share the motivation that librosa has for this -- we're not building an audio analyzer so it'd be safer not to set one. (same for `sr=22050` in `melspectrogram`, will change it later.)
@ksanjeevan @faroit @f0k

### CODE REVIEW

File: torchaudio_contrib/functional.py [hunk 1]
```diff
@@ -46,60 +46,51 @@ def _hertz_to_mel(hz, htk):
                        torch.log(hz / min_log_hz) / logstep, mel)
 
 
-def stft(waveforms, fft_len, hop_len, window,
-         pad=0, pad_mode="reflect", **kwargs):
+def stft(waveforms, fft_len, hop_len=None, win_len=None, window=None,
+         center=True, pad_mode='reflect', normalized=False, onesided=True):
```
Comment (ksanjeevan) [inline torchaudio_contrib/functional.py]: If we're just wrapping `torch.stft` to allow for multichannel, do we have to explicitly pass the parameters? Can we have something like:
```python
def stft(waveforms, **stft_kwargs):
    ... # mutli-channel
    complex_specgrams = torch.stft(waveforms, **stft_kwargs)
    return complex_specgrams
```
Then if there were any changes to `torch.stft` we wouldn't be explicitly out of sync...?

File: torchaudio_contrib/layers.py [hunk 2]
```diff
@@ -42,39 +42,43 @@ class STFT(_ModuleNoStateBuffers):
...
+        self.center = center
         self.pad_mode = pad_mode
-        self.kwargs = kwargs
+        self.normalized = normalized
+        self.onesided = onesided
 
         self.register_buffer('window', window)
 
-    def _stft_defaults(self, fft_len, hop_len, frame_len, window):
+    def _stft_defaults(self, fft_len, hop_len, win_len, window):
```
Comment (ksanjeevan) [inline torchaudio_contrib/layers.py]: This should be deleted since we're no longer handling defaults.

File: torchaudio_contrib/functional.py (hunk 1 above)
Comment (keunwoochoi) [reply torchaudio_contrib/functional.py]: Sounds good. I wanted to make things more explicit, but probably a little more verbose docstring would do that.

File: torchaudio_contrib/layers.py [hunk 3]
```diff
@@ -42,39 +42,43 @@ class STFT(_ModuleNoStateBuffers):
...
-    def __init__(self, fft_len=2048, hop_len=None, frame_len=None,
-                 window=None, pad=0, pad_mode="reflect", **kwargs):
+    def __init__(self, fft_len, hop_len=None, win_len=None,
+                 window=None, center=True, pad_mode='reflect',
+                 normalized=False, onesided=True):
 
         super(STFT, self).__init__()
 
         # Get default values, window so it can be registered as buffer
         self.fft_len, self.hop_len, window = self._stft_defaults(
```
Comment (ksanjeevan) [inline torchaudio_contrib/layers.py]: Also to remove!

**Comment by keunwoochoi** [comment]
Ohh. If we wrap `torch.stft` with `**kwargs`, we can't use the names we decided in #36. We can keep `fft_len` because it's a positional argument, but can't use `hop_len` and `win_len`. Instead, we have to use 
`hop_length` and `win_length`. But, if so, `fft_len` wouldn't be consistent (why `len` and not `length` here?).

To keep consistency, we should also adopt `n_fft`. 

..which is not in the `num_` format, which we were using.

Ugh.

**Comment by keunwoochoi** [comment]
Probably we should discard `**kwargs` idea, which is also not a bad idea because
- practically, i don't think the default values of `torch.stft` would change often. maybe once at maximum?
- for `STFT` layer, it's a good thing because then we can keep the values explicitly as attributes

**Comment by ksanjeevan** [comment]
> Ohh. If we wrap torch.stft with **kwargs, we can't use the names we decided in #36. 

Right...

> Probably we should discard **kwargs idea, which is also not a bad idea because

Fair.
